Formula/gradle.rb

```diff
@@ -12,14 +12,16 @@ class Gradle < Formula
 
   bottle :unneeded
 
+  depends_on :java => "1.6+"
+
   def install
     libexec.install %w[bin lib]
-    bin.install_symlink libexec+"bin/gradle"
+    (bin/"gradle").write_env_script libexec/"bin/gradle",
+      Language::Java.java_home_env.merge(:GRADLE_HOME => libexec)
```

Comment:
This change breaks homebrew-installed `gradle`. At the least, it causes `gradle` to no longer respect my `JAVA_HOME`. More specifically, if I have JDK8 and JDK9-ea builds installed and point my `JAVA_HOME` to the JDK8 build for everyday development, but `gradle` will now pick up the JDK9-ea install.

```
21:30:42 ⌁49% [jason:~] $ echo $JAVA_HOME
/Library/Java/JavaVirtualMachines/jdk1.8.0_92.jdk/Contents/Home
21:30:45 ⌁49% [jason:~] $ gradle --version

------------------------------------------------------------
Gradle 2.13
------------------------------------------------------------

Build time:   2016-04-25 04:10:10 UTC
Build number: none
Revision:     3b427b1481e46232107303c90be7b05079b05b1c

Groovy:       2.4.4
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          9-ea (Oracle Corporation 9-ea+120)
OS:           Mac OS X 10.11.5 x86_64

21:30:48 ⌁49% [jason:~] $ 
```

Of course, this is not what I want, and `gradle` is broken on JDK9-ea builds right now.

Moreover, this change is unnecessary. The [built-in gradle script will already try to detect what `JAVA_HOME` should be *if* it is not set, but it will respect what you have set if you do have it set](https://github.com/gradle/gradle/blob/65b309141421f4aaeae35d7eab830bee9ea4b790/subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt#L70-L89). This changes completely breaks that.

Finally, looking at #889, it appears this was done for `GRADLE_HOME`. But having `GRADLE_HOME` set is completely unnecessary, and [does not do anything useful](https://github.com/gradle/gradle/search?q=GRADLE_HOME&type=Code&utf8=✓). Since `/usr/local/bin` is on the path, there's no need to set `GRADLE_HOME` and add `$GRADLE_HOME/bin` to the path.

At the least, setting `JAVA_HOME` and `GRADLE_HOME` should be wrapped in conditionals. But I think that this change should just be reverted. I opened #1650.